### PR TITLE
Caching abstraction - Profile Fragment

### PR DESCRIPTION
**Pull request by lymichelle21**
New stuff:
- Implemented MVVM architecture to profile fragment where view model gets bookmarks and determines if bookmarked event is in cache or needs to be called from API
- Made populate event code more reusable across classes with helper class SetEventInfo

Notes:
- May do MVVM part 3 with recommendation fragment later on

### CODE REVIEW

File: app/src/main/java/com/capstone/event_finder/network/EventViewModel.java [hunk 1]
```diff
@@ -40,8 +48,57 @@ public void refreshEvents() {
...
 
-    public LiveData<List<Event>> eventInCache(String eventId) {
-        return eventRepository.eventInCache(eventId);
+    public LiveData<List<Event>> getBookmarks(ParseUser user) {
+        JsonArray allBookmarks = new JsonArray();
+        queryUserBookmarksFromParse(allBookmarks, user);
+        return bookmarkList;
+    }
+
+    private void tryRetrieveEventInCache(String eventId, JsonArray allBookmarks, List<Event> bookmarks) {
```
Comment (erikweissfb) [inline app/src/main/java/com/capstone/event_finder/network/EventViewModel.java:57]: 💡 Nit: I'm not sure we need either this `allBookmarks` list nor this new `bookmarks` list.

We can probably just update the live data `bookmarkList` directly as we go.

**Review by erikweissfb — APPROVED** [review]
Overall, this looks great.  I love PRs that delete more code than they create!

Some highlights are:
* Reducing duplicate event object creation code in 3 places with your new `SetEventInfo` utility.
* Moving the handling of building the bookmarked event list to the View Model.

File: app/src/main/java/com/capstone/event_finder/network/EventApi.java [hunk 2]
```diff
@@ -51,23 +52,21 @@ private Collection<? extends Event> convertToList(JsonObject result) {
         for (int i = 0; i < events.size(); i++) {
             JsonObject temp = (JsonObject) events.get(i);
             Event event = new Event();
-            populateEventInfo(event, temp);
+            SetEventInfo.populateEvent(event, temp);
```
[the remark this answers is not in the document]
Comment (lymichelle21) [reply app/src/main/java/com/capstone/event_finder/network/EventApi.java:55]: Yes! Updated in next PR

File: app/src/main/java/com/capstone/event_finder/network/EventViewModel.java [hunk 3]
```diff
@@ -40,8 +48,57 @@ public void refreshEvents() {
...
+                    bookmarkList.postValue(bookmarks);
+                });
+            } else {
+                bookmarks.addAll(events);
+            }
+            bookmarkList.postValue(bookmarks);
+        });
+    }
+
+    public void queryUserBookmarksFromParse(JsonArray allBookmarks, ParseUser user) {
```
[the remark this answers is not in the document]
Comment (lymichelle21) [reply app/src/main/java/com/capstone/event_finder/network/EventViewModel.java:71]: Good catch - fixing in next PR

File: app/src/main/java/com/capstone/event_finder/network/EventViewModel.java [hunk 4]
```diff
@@ -40,8 +48,57 @@ public void refreshEvents() {
...
+        ParseQuery<Bookmark> query = ParseQuery.getQuery(Bookmark.class);
+        query.whereEqualTo(Bookmark.KEY_USER, user);
+        query.setLimit(POST_LIMIT);
+        query.addDescendingOrder("createdAt");
+        query.findInBackground((bookmarks, e) -> {
+            if (e != null) {
+                e.printStackTrace();
+                return;
+            }
+            userBookmarks.clear();
```
[the remark this answers is not in the document]
Comment (lymichelle21) [reply app/src/main/java/com/capstone/event_finder/network/EventViewModel.java:82]: Good to know - fixed in next PR!

File: app/src/main/java/com/capstone/event_finder/network/EventViewModel.java (hunk 1 above)
Comment (lymichelle21) [reply app/src/main/java/com/capstone/event_finder/network/EventViewModel.java:57]: Check PR comment above

**Pull request merged**